Assets/Scripts/InfiniteMana.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/*********************************************************************************************
+ * Script is meant to be applied to a Infinite Mana Spawner
+ * Inherits the entity class in order for the object to appear to take damage
+ * When TakeDamage is called, 1 - 3 mana orbs will spawn which can be picked up by the player
+ * Spawner will always regen back to full health which makes it indestructible.
+*********************************************************************************************/
+public class InfiniteMana : Entity
+{
+    public GameObject manaOrb; //Set this to the Mana Orb prefab
+    private float maxHealth; 
+    private float spawnerCordsX; 
+    private float spawnerCordsY;
+    private static System.Random rand = new System.Random();
+
+    public override void TakeDamage(float baseDamage)
+    {
+        base.TakeDamage(baseDamage);
```

Comment:
getting rid of this line will prevent its health from going down so that you don't need to set its health to something so high